src/Microsoft.AspNetCore.OData/Batch/ODataBatchReaderExtensions.cs

```diff
@@ -227,6 +243,22 @@ private static HttpContext CreateHttpContext(HttpContext originalContext)
             context.Request.Cookies = originalContext.Request.Cookies;
             foreach (KeyValuePair<string, StringValues> header in originalContext.Request.Headers)
             {
+                string headerKey = header.Key.ToLowerInvariant();
+                // do not copy over headers that should not be inherited from batch to individual requests
+                if (nonInheritableHeaders.Contains(headerKey))
+                {
+                    continue;
+                }
+                // some preferences may be inherited, others discarded
+                if (headerKey == "prefer")
+                {
+                    string preferencesToInherit = GetPreferencesToInheritFromBatch(header.Value);
+                    if (!string.IsNullOrEmpty(preferencesToInherit))
+                    {
+                        context.Request.Headers.Add(header.Key, preferencesToInherit);
+                    }
+                    continue;
+                }
                 context.Request.Headers.Add(header);
```

Comment:
I think it's a bit easier to read without the continues:

            if (!nonInheritableHeaders.Contains(headerKey))
            {
                // some preferences may be inherited, others discarded
                if (headerKey == "prefer")
                {
                    string preferencesToInherit = GetPreferencesToInheritFromBatch(header.Value);
                    if (!string.IsNullOrEmpty(preferencesToInherit))
                    {
                        context.Request.Headers.Add(header.Key, preferencesToInherit);
                    }
                }
                else
                {
                    context.Request.Headers.Add(header);
                }
            }